packages/components/menu/src/Menu.tsx

```diff
@@ -150,12 +150,26 @@ export function Menu(props: MenuProps) {
     [closeAndFocusTrigger, handleArrowsKeyDown],
   );
 
+  const isMouseDown = useRef<Boolean>(false);
```

Comment:
```suggestion
 // the comment that explains the issue in safari that we are trying to fix here. Because the one we have in the `onMouseDown` event is not clear enough for me. 
  const isTriggerMouseDown = useRef<Boolean>(false);
```
WDYT?